docs/howto/composite-primary-key.txt

```diff
@@ -0,0 +1,159 @@
+
+.. versionadded:: 5.2
+
+In Django, each model has a primary key. By default, this primary key consists
+of a single field.
+
+In most cases, a single primary key should suffice. In database design,
+however, defining a primary key consisting of multiple fields is sometimes
+necessary.
+
+To define a composite primary key, set the model's ``pk`` to an instance of
+``CompositePrimaryKey``:
+
+.. code-block:: pycon
+
+    class Product(models.Model):
+        name = models.CharField(max_length=100)
+
+    class Order(models.Model):
+        reference = models.CharField(max_length=20, primary_key=True)
+
+    class OrderLineItem(models.Model):
+        pk = models.CompositePrimaryKey("product_id", "order_id")
+        product = models.ForeignKey(Product, on_delete=models.CASCADE)
+        order = models.ForeignKey(Order, on_delete=models.CASCADE)
+        quantity = models.IntegerField()
+
+This will instruct Django to create a composite primary key
+(``PRIMARY KEY (product_id, order_id)``) when creating the table.
+
+Composite primary keys are represented with ``tuple``\s:
+
+.. code-block:: pycon
+
+    >>> product = Product.objects.create(name="apple")
+    >>> order = Order.objects.create(reference="A755H")
+    >>> item = OrderLineItem.objects.create(
+    ...     product=product, order=order, quantity=1
+    ... )
+    >>> item.pk
+    (1, "A755H")
+
+You can assign a ``tuple`` to a composite primary key. This sets the associated
+field values.
+
+.. code-block:: pycon
+
+    >>> item = OrderLineItem(pk=(2, "B142C"))
+    >>> item.pk
+    (2, "B142C")
+    >>> item.product_id
+    2
+    >>> item.order_id
+    "B142C"
+
+Or filter composite ``pk``\s by ``tuple``\s:
+
+.. code-block:: pycon
+
+    >>> OrderLineItem.objects.filter(pk=(1, "A755H")).count()
+    1
+
+Migrating to a composite primary key
+====================================
+
+Django doesn't support migrating to, or from, a composite primary key after the
+table is created.
+
+If you would like to migrate an existing table from a single primary key to a
+composite primary key, follow your database backend's instructions to do so.
+
+Once the composite primary key is in place, add the ``CompositePrimaryKey``
+field to your model. This allows Django to recognize and handle the composite
+primary key appropriately.
+
+While migration operations (e.g. ``AddField``, ``AlterField``) on
+primary key fields are not supported, ``makemigrations`` will still detect
+changes.
+
+In order to avoid errors, it's recommended to apply such migrations with
+``--fake``.
+
+Alternatively, :class:`.SeparateDatabaseAndState` may be used to execute
+the backend-specific migrations and Django-generated migrations in
+a single operation.
+
+How to define a composite foreign key
+=====================================
+
+At the moment, relationship fields (``ForeignKey``, ``OneToOneField``,
+``ManyToManyField``) don't support composite primary keys.
+
+So, given the ``OrderLineItem`` model, we cannot do:
```

Comment:
```suggestion
Composite primary keys and relations
====================================

:ref:`Relationship fields <relationship-fields>`, including
:ref:`generic relations <generic-relations>` do not support composite primary
keys.

For example, given the ``OrderLineItem`` model, we cannot do:
```
(for this suggestion to work, I think we need to add a `.. _relationship-fields` link)

I think let's include the mention of generic relations here and remove the `Composite primary keys and generic relations` (as I don't think it deserves it's own separate section)